nativepython/type_wrappers/alternative_wrapper.py

```diff
@@ -71,6 +72,42 @@ def convert_copy_initialize(self, context, target, toStore):
             target.expr.store(toStore.nonref_expr)
         )
 
+    def convert_to_type_with_target(self, context, e, targetVal, explicit):
+        # there is deliberately no code path for "not explicit" here
+        # Alternative conversions must be explicit
+        assert targetVal.isReference
+
+        target_type = targetVal.expr_type
+
+        if target_type.typeRepresentation == Bool:
+            alt = e.expr_type.typeRepresentation
+            if hasattr(alt.__bool__, "__typed_python_category__") and alt.__bool__.__typed_python_category__ == 'Function':
+                assert len(alt.__bool__.overloads) == 1
+                y = context.call_py_function(alt.__bool__.overloads[0].functionObj, (e,), {})
+                if y is None:
+                    return context.constant(False)
+
+                ret_type = y.expr_type.typeRepresentation
```

Comment:
it might return 'object' here, which we then need to try to convert to bool. So I think you could get away with

```
return y.expr_type.convert_to_type_with_target(context, y, targetVal, False)
```

Reply:
Yes, that works!